gremlin-scala/src/main/scala/gremlin/scala/GremlinScala.scala

```diff
@@ -364,6 +364,18 @@ case class GremlinScala[End, Labels <: HList](traversal: GraphTraversal[_, End])
     GremlinScala[A, Labels](traversal.union(mappedTraversals: _*))
   }
 
+  def choose[A, B](
+    predicate: GremlinScala[End, HNil] ⇒ GremlinScala[A, _],
+    trueChoice: GremlinScala[End, HNil] ⇒ GremlinScala[B, _],
+    falseChoice: GremlinScala[End, HNil] ⇒ GremlinScala[B, _]): GremlinScala[B, Labels] = {
+    val p: Traversal[_, A] = predicate(start).traversal
```

Comment:
I thought the same thing too, but when I looked at the TinkerPop docs, it says that the predicate is considered `true` if the traversal is non-empty.

What's your architecture leaning here: preserve the original Gremlin/TinkerPop semantics, or adapt them to make them more type-consistent?